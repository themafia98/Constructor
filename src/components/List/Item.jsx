import React, {useState} from 'react';

function Item(props) {
    const [id] = useState(props.id);
    const [item] = useState({...props.item});
    return (
        <div className = 'item' data-id = {id}>{item.name}</div>
    )
}

export default Item;