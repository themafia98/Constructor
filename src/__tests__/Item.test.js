import React from 'react';
import { MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import Events from 'events';
import Item from '../components/List/Item';
import firebase from '../Firebase/Firebase';

let ItemComponents = renderer.create(
    <MemoryRouter initialEntries = {['/Cabinet/Build/0']}>
            <Item firebase = {firebase} name ='super' 
                cabinetStream = {new Events()} 
                id = {0} 
                title = {'test1'} 
                type = {"portfolio"}  />
    </MemoryRouter>
)


const act = renderer.act;

test('Item.js test ', function() {

    let snapshotItemComponents = ItemComponents.toJSON();
    expect(snapshotItemComponents).toMatchSnapshot();

    const buttonEnter = ItemComponents.root.find(buttonEnter => buttonEnter.props.className === 'projectControllerButton_enter');
    const buttonDelete = ItemComponents.root.find(buttonDelete => buttonDelete.props.className === 'projectControllerButton_delete');

    act(()=> {
        buttonEnter.props.onClick({stopPropagation: () => null});
    });


    snapshotItemComponents = ItemComponents.toJSON();
    expect(snapshotItemComponents).toMatchSnapshot();

    act(()=>{
        buttonDelete.props.onClick({stopPropagation: () => null});
    });

    snapshotItemComponents = ItemComponents.toJSON();
    expect(snapshotItemComponents).toMatchSnapshot();
});