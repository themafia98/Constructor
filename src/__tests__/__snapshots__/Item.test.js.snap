// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item.js test  1`] = `
<div
  className="item"
  data-id={0}
>
  <p
    className="ProjectName"
  >
    super
  </p>
  <p
    className="ProjectType"
  >
    portfolio
  </p>
  <div
    className="projectController"
  >
    <input
      className="projectControllerButton_enter"
      onClick={[Function]}
      type="button"
      value="enter build"
    />
    <input
      className="projectControllerButton_delete"
      onClick={[Function]}
      type="button"
      value="delete"
    />
  </div>
  <input
    className="productionButton"
    onClick={[Function]}
    type="button"
    value="Production"
  />
</div>
`;

exports[`Item.js test  2`] = `
<div
  className="item"
  data-id={0}
>
  <p
    className="ProjectName"
  >
    super
  </p>
  <p
    className="ProjectType"
  >
    portfolio
  </p>
  <div
    className="projectController"
  >
    <input
      className="projectControllerButton_enter"
      onClick={[Function]}
      type="button"
      value="enter build"
    />
    <input
      className="projectControllerButton_delete"
      onClick={[Function]}
      type="button"
      value="delete"
    />
  </div>
  <input
    className="productionButton"
    onClick={[Function]}
    type="button"
    value="Production"
  />
</div>
`;

exports[`Item.js test  3`] = `
<div
  className="item"
  data-id={0}
>
  <p
    className="ProjectName"
  >
    super
  </p>
  <p
    className="ProjectType"
  >
    portfolio
  </p>
  <div
    className="projectController"
  >
    <input
      className="projectControllerButton_enter"
      onClick={[Function]}
      type="button"
      value="enter build"
    />
    <input
      className="projectControllerButton_delete"
      onClick={[Function]}
      type="button"
      value="delete"
    />
  </div>
  <input
    className="productionButton"
    onClick={[Function]}
    type="button"
    value="Production"
  />
</div>
`;
